[2024-03-19 16:48:51,437] {taskinstance.py:1032} INFO - Dependencies all met for <TaskInstance: dag_ml_life_cycle.evaluate_model scheduled__2024-03-19T16:46:08.348952+00:00 [queued]>
[2024-03-19 16:48:51,448] {taskinstance.py:1032} INFO - Dependencies all met for <TaskInstance: dag_ml_life_cycle.evaluate_model scheduled__2024-03-19T16:46:08.348952+00:00 [queued]>
[2024-03-19 16:48:51,448] {taskinstance.py:1238} INFO - 
--------------------------------------------------------------------------------
[2024-03-19 16:48:51,448] {taskinstance.py:1239} INFO - Starting attempt 1 of 4
[2024-03-19 16:48:51,448] {taskinstance.py:1240} INFO - 
--------------------------------------------------------------------------------
[2024-03-19 16:48:51,456] {taskinstance.py:1259} INFO - Executing <Task(_PythonDecoratedOperator): evaluate_model> on 2024-03-19 16:46:08.348952+00:00
[2024-03-19 16:48:51,463] {standard_task_runner.py:52} INFO - Started process 60609 to run task
[2024-03-19 16:48:51,469] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'dag_ml_life_cycle', 'evaluate_model', 'scheduled__2024-03-19T16:46:08.348952+00:00', '--job-id', '880', '--raw', '--subdir', 'DAGS_FOLDER/lifecycle_ml_example/dags/dag_ml_life_cycle.py', '--cfg-path', '/var/folders/t4/c2kdpvnj6r793w3k056dkc640000gp/T/tmphu92edz1', '--error-file', '/var/folders/t4/c2kdpvnj6r793w3k056dkc640000gp/T/tmpef7pw257']
[2024-03-19 16:48:51,472] {standard_task_runner.py:77} INFO - Job 880: Subtask evaluate_model
[2024-03-19 16:48:51,519] {logging_mixin.py:109} INFO - Running <TaskInstance: dag_ml_life_cycle.evaluate_model scheduled__2024-03-19T16:46:08.348952+00:00 [running]> on host 1.0.0.127.in-addr.arpa
[2024-03-19 16:48:51,550] {taskinstance.py:1424} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=dag_ml_life_cycle
AIRFLOW_CTX_TASK_ID=evaluate_model
AIRFLOW_CTX_EXECUTION_DATE=2024-03-19T16:46:08.348952+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2024-03-19T16:46:08.348952+00:00
[2024-03-19 16:48:51,551] {dag_ml_life_cycle.py:93} INFO - Getting gold data from ml_lifecycle_airflow/in-vehicle-coupon-recommendation_gold.csv ...
[2024-03-19 16:49:01,733] {dag_ml_life_cycle.py:105} INFO - Downloading model from ml_lifecycle_airflow/artifacts/model.pkl ...
[2024-03-19 16:49:04,487] {logging_mixin.py:109} INFO - File artifacts/model.pkl downloaded in /tmp/model.pkl.
[2024-03-19 16:49:04,496] {training.py:111} INFO - Model loaded from /tmp/model.pkl
[2024-03-19 16:49:04,497] {dag_ml_life_cycle.py:112} INFO - Evaluating model ...
[2024-03-19 16:49:04,624] {logging_mixin.py:109} INFO - accuracy (test): 0.7425323817076395
[2024-03-19 16:49:04,632] {logging_mixin.py:109} INFO - precision (test): 0.7490297542043984
[2024-03-19 16:49:04,633] {logging_mixin.py:109} INFO - recall (test): 0.8158759981211836
[2024-03-19 16:49:04,633] {logging_mixin.py:109} INFO - roc-auc (train-proba): 0.9961175563723904
[2024-03-19 16:49:04,634] {logging_mixin.py:109} INFO - roc-auc (test-proba): 0.8127045016053429
[2024-03-19 16:49:04,637] {dag_ml_life_cycle.py:119} INFO - Uploading metrics to ml_lifecycle_airflow/in-vehicle-coupon-recommendation_gold.csv ...
[2024-03-19 16:49:06,688] {local_task_job.py:191} WARNING - The recorded hostname 1.0.0.127.in-addr.arpa does not match this instance's hostname macbook-pro-2.local
[2024-03-19 16:49:06,709] {process_utils.py:120} INFO - Sending Signals.SIGTERM to group 60609. PIDs of all processes in the group: [60609]
[2024-03-19 16:49:06,710] {process_utils.py:75} INFO - Sending the signal Signals.SIGTERM to group 60609
[2024-03-19 16:49:06,710] {taskinstance.py:1408} ERROR - Received SIGTERM. Terminating subprocesses.
[2024-03-19 16:49:06,719] {taskinstance.py:1700} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1329, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1455, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1511, in _execute_task
    result = execute_callable(context=context)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/decorators/base.py", line 134, in execute
    return_value = super().execute(context)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/operators/python.py", line 174, in execute
    return_value = self.execute_callable()
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/operators/python.py", line 185, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/Users/macbook/Desktop/airflow_tutorial/airflow/dags/lifecycle_ml_example/dags/dag_ml_life_cycle.py", line 120, in evaluate_model_task
    upload_df_to_gcs(
  File "/Users/macbook/Desktop/airflow_tutorial/airflow/dags/lifecycle_ml_example/src/utils/f_gcs_storage.py", line 31, in upload_df_to_gcs
    df.to_csv(csv_buffer, index=False)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/util/_decorators.py", line 211, in wrapper
    return func(*args, **kwargs)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/core/generic.py", line 3720, in to_csv
    return DataFrameRenderer(formatter).to_csv(
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/util/_decorators.py", line 211, in wrapper
    return func(*args, **kwargs)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/io/formats/format.py", line 1189, in to_csv
    csv_formatter.save()
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/io/formats/csvs.py", line 261, in save
    self._save()
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/io/formats/csvs.py", line 266, in _save
    self._save_body()
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/io/formats/csvs.py", line 304, in _save_body
    self._save_chunk(start_i, end_i)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/io/formats/csvs.py", line 314, in _save_chunk
    ix = self.data_index[slicer]._format_native_types(**self._number_format)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/core/indexes/numeric.py", line 285, in _format_native_types
    from pandas.io.formats.format import FloatArrayFormatter
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1410, in signal_handler
    raise AirflowException("Task received SIGTERM signal")
airflow.exceptions.AirflowException: Task received SIGTERM signal
[2024-03-19 16:49:06,739] {taskinstance.py:1267} INFO - Marking task as UP_FOR_RETRY. dag_id=dag_ml_life_cycle, task_id=evaluate_model, execution_date=20240319T164608, start_date=20240319T214851, end_date=20240319T214906
[2024-03-19 16:49:06,750] {standard_task_runner.py:89} ERROR - Failed to execute job 880 for task evaluate_model
Traceback (most recent call last):
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/task/task_runner/standard_task_runner.py", line 85, in _start_by_fork
    args.func(args, dag=self.dag)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/cli/cli_parser.py", line 48, in command
    return func(*args, **kwargs)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/utils/cli.py", line 92, in wrapper
    return f(*args, **kwargs)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 298, in task_run
    _run_task_by_selected_method(args, dag, ti)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 107, in _run_task_by_selected_method
    _run_raw_task(args, ti)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 180, in _run_raw_task
    ti._run_raw_task(
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/utils/session.py", line 70, in wrapper
    return func(*args, session=session, **kwargs)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1329, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1455, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1511, in _execute_task
    result = execute_callable(context=context)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/decorators/base.py", line 134, in execute
    return_value = super().execute(context)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/operators/python.py", line 174, in execute
    return_value = self.execute_callable()
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/operators/python.py", line 185, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/Users/macbook/Desktop/airflow_tutorial/airflow/dags/lifecycle_ml_example/dags/dag_ml_life_cycle.py", line 120, in evaluate_model_task
    upload_df_to_gcs(
  File "/Users/macbook/Desktop/airflow_tutorial/airflow/dags/lifecycle_ml_example/src/utils/f_gcs_storage.py", line 31, in upload_df_to_gcs
    df.to_csv(csv_buffer, index=False)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/util/_decorators.py", line 211, in wrapper
    return func(*args, **kwargs)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/core/generic.py", line 3720, in to_csv
    return DataFrameRenderer(formatter).to_csv(
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/util/_decorators.py", line 211, in wrapper
    return func(*args, **kwargs)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/io/formats/format.py", line 1189, in to_csv
    csv_formatter.save()
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/io/formats/csvs.py", line 261, in save
    self._save()
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/io/formats/csvs.py", line 266, in _save
    self._save_body()
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/io/formats/csvs.py", line 304, in _save_body
    self._save_chunk(start_i, end_i)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/io/formats/csvs.py", line 314, in _save_chunk
    ix = self.data_index[slicer]._format_native_types(**self._number_format)
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/pandas/core/indexes/numeric.py", line 285, in _format_native_types
    from pandas.io.formats.format import FloatArrayFormatter
  File "/Users/macbook/anaconda3/envs/airflow_env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1410, in signal_handler
    raise AirflowException("Task received SIGTERM signal")
airflow.exceptions.AirflowException: Task received SIGTERM signal
[2024-03-19 16:49:06,768] {process_utils.py:70} INFO - Process psutil.Process(pid=60609, status='terminated', exitcode=1, started='16:48:51') (60609) terminated with exit code 1
